### ISSUE TITLE
Missing Resource on How to use Doxygen for "Normal Documentation"

### ISSUE THREAD

**Issue by cuppajoeman** [doxygen/doxygen#10418]
**Describe the bug**
On the main page of the site, I was informed that doxygen can help me in 3 ways. The third way is this:

>  You can also use doxygen for creating normal documentation (as I did for the doxygen user manual and web-site). 

I headed over to the documentation to read about this, and to my suprise I wasn't able to find any information on how I can make normal documentation. 

**Expected behavior**
I expected to easily see how to make normal documentation

**To Reproduce**
Go to the main site and try to figure out how to create normal documentation

**Notes**
Doing some searching online I found this SO question: https://stackoverflow.com/questions/9502426/how-to-make-an-introduction-page-with-doxygen which helps a bit.

I looked in the doxyfile for this repository and and looks like [this line](https://github.com/doxygen/doxygen/blob/e29c4a5e86941ed6b6c2437fcdf82263a2679268/doc/Doxyfile#L38) seems to provide many of the pages of the site

But the most useful thing as of yet has been looking in https://github.com/doxygen/doxygen/blob/master/doc/index.doc.

I found the last word of the [main page](https://www.doxygen.nl/index.html) (available) and then found the corresponding place in the index.doc file [here](https://github.com/doxygen/doxygen/blob/e29c4a5e86941ed6b6c2437fcdf82263a2679268/doc/index.doc#L58) , after that line we can see that there's a newline and then the source for the [documentation](https://www.doxygen.nl/manual/index.html) tab starts.

If there is really no resource available on the main site that tells us how to create normal documentation, I was hoping the creator or someone could help give some tips on how we can do this and then eventually add it to the main site.

**Comment by cuppajoeman** [comment]
After using doxygen for a bit and figuring out how things work, your response makes more sense, but I still have a few points.

One example of creating "normal documentation" might be a usage/tutorial/examples section which might explain how to use the documented functions together to create sample programs, this may also incorporate adding custom elements to the navbar to grab the user's attention.

Even though it might be obvious now how normal documentation works, I think my younger self would have appreciated a section dedicated on how to actually do the relevant tasks that come with creating normal documentation.

Also taking a look online for questions/resources relating to this topic, there are some:

* https://stackoverflow.com/questions/3052036/how-to-include-custom-files-in-doxygen
* http://entrenchant.blogspot.com/2009/09/doxygen-gotchas.html
* https://stackoverflow.com/questions/9502426/how-to-make-an-introduction-page-with-doxygen

Which should at least indicate some other people might appreciate this as well.

---

Here's my proposal. Within the doxygen manual we add a section called "custom pages", where we write something along the lines of:

Doxygen can be used to create custom pages that are not part of the API of your library/program, the purpose of these pages is to enrich your documentation with anything else that you think the user may find useful.

To create custom pages, use one of the supported the file formats: .dox, .doc, .txt, .md. Inside such a file doxygen will treat it like any other source file, so one option could be to consider this file as large single doxygen comment.

`manual/index.doc`:
```
/** \mainpage My Library Manual
- Building
- Basics
- Examples
*/
```

You'll note that the `\mainpage` command was used, which tells doxygen to use this page as, well, the main page. For other pages, preface them with the `\page` command.

By default doxygen will not know about these custom files so we'll need to let it know through the `INPUT` attribute in our Doxyfile, that is with our current configuration, would have this line in our Doxyfile:

```
INPUT                  = manual.doc
```

Next, we want to add the instructions on how to build the project, so we create `manual/building/index.doc`, we've read a bit more of the documentation and [found out](https://www.doxygen.nl/manual/htmlcmds.html) that doxygen supports a subset of html's elements, so we write the following:

```
/** \page Building

<h2>linux</h2>
<p>
    A simple way to build this project is with cmake, clone the  repository, cd into the root of the project and run:
</p>

<pre>
    <code>
        mkdir my_build
        cmake -S . -B my_build
        cd my_build
        cmake --build .
    </code>
</pre>

*/
```

At this point we could now go ahead and add `manual/building/index.doc` to our INPUT's with comma separation, but this might become annoying over time as we build up our manual, instead we'll just change it reference our manual folder:

```
INPUT                  = manual/
```

And set 

```
RECURSIVE              = YES
```

As your manual scales up, you might want to also have a nice tree view to show you where you are in the manual to stay organized. This is easy enough to set up, turn it on with 

```
GENERATE_TREEVIEW      = YES
```
In your doxyfile.

If you have multiple files linked together using the `\ref` command then you may notice that your tree is more like a pile of leaves. To remedy this  check out [subpaging](https://www.doxygen.nl/manual/grouping.html#subpaging).

This disucssion should give you some direction on how to build a scalable manual to enrich your documentation, take this and modify it to suit your needs.

**Comment by albert-github** [comment]
As a side node did you also look at the chapter "Customizing the output" (https://www.doxygen.nl/manual/customize.html) and especially the part "Changing the layout of pages" (https://www.doxygen.nl/manual/customize.html#layout),also search for `type="user"` on that page.

**Comment by doxygen** [comment]
With the latest commit, the following should also work to have a collection of markdown pages, which are shown as a hierarchy in the tree view, while the content is still collected on a single page.

```
INPUT                  = main.md
USE_MDFILE_AS_MAINPAGE = main.md
```

main.md
```md
# The main page

[TOC]

@section topic1 First main topic
@include{doc} topic1.md
@section topic2 Second main topic
@include{doc} topic2.md
@section snippet1 First snippet
@snippet{doc} snippets.md SNIPPET1
@section snippet2 Second snippet
@snippet{doc} snippets.md SNIPPET2

Closing words
```

topic1.md
```md
Topic 1 explanation.

@subsection st1 First subtopic
@include{doc} subtopic1.md
@subsection st2 Second subtopic
@include{doc} subtopic2.md

More info about topic 1
```

topic2.md
```md
Topic 2 explanation.
```

subtopic1.md
```md
A subtopic
```

subtopic2.md
```md
another sub topic
```

snippets.md
```md
[SNIPPET1]
Hello this is a bold **snippet**.
[SNIPPET1]

[SNIPPET2]
Hello this is an italic _snippet_.
[SNIPPET2]
```

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
